client/src/pages/Signup.js

```diff
@@ -95,6 +97,17 @@ const Signup = () => {
             register(username, password, email, (data) => {
                 // onSucess
                 console.log("SUCCESS")
+                // get profile and update context
+                getProfile((data) => {
+                    setUser({
+                        username: data.username,
+                        icon: (data.icon == null) ? process.env.PUBLIC_URL + 'images/avatar-1.png' : data.icon,
```

Comment:
when you sign up, you won't have an icon so we can probably just assume we won't get anything in return